larq/callbacks.py

```diff
@@ -68,3 +71,47 @@ def on_train_begin(self, logs=None):
 
     def on_epoch_end(self, epoch, logs=None):
         self._maybe_log_and_store(self.epoch_previous_weights, logs)
+
+
+class BopLearningRateScheduler(tf.keras.callbacks.Callback):
```

Comment:
I don't think we should add a new callback just dedicated for Bop. I think it would be could to find ways of handling this usecase inside the Bop class.